scripts/full_check.sh

```diff
@@ -0,0 +1,111 @@
+#! /bin/bash
+
+# Copyright (C) 2024 Intel Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+# this is for developer local check only.
+# not used by github action
+
+check_tool() {
+  which $1 &> /dev/null || (echo "$1 not found!" && exit 1)
+}
+
+for arg in "$@"; do
+  case $arg in
+    -f|--format)
+      CHECK_FORMAT=1
+      ;;
+    -t|--tidy)
+      CHECK_TIDY=1
+      ;;
+    -l|--license)
+      CHECK_LICENSE=1
+      ;;
+    -c|--clean)
+      CLEANUP=1
+      ;;
+    *) # Handle other unknown options
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+# set PR_REF if your pr target is not main
+: ${PR_REF:=main}
+
+PROJECT_ROOT="$(cd "$(dirname "$0")" && pwd)/../"
+cd "$PROJECT_ROOT"
+
+MERGE_BASE=$(git merge-base $PR_REF HEAD)
+CHANGED_FILES=$(git diff --name-only $MERGE_BASE HEAD)
+
+# if you do not have clang/clang++/clang-tidy/clang-format
+# please install it
+# conda install -c conda-forge clangxx cxx-compiler clang-tools
+
+if [ -n "$CHECK_TIDY" ]; then
+  echo "start tidy check..."
+  if [ -z "${MLIR_DIR}" ]; then
+    echo "The environment variable MLIR_DIR is not set."
+    exit 1
+  fi
+  check_tool "clang"
+  check_tool "clang++"
+  check_tool "clang-tidy"
+  check_tool "lit"
+
+  TIDY_ROOT="${PROJECT_ROOT}/build/tidy"
+  [ -n "$CLEANUP" ] && rm -rf "${TIDY_ROOT}"
+  mkdir -p "${TIDY_ROOT}"
```

Comment:
```suggestion
  [ -d "${TIDY_ROOT}" ] || mkdir -p "${TIDY_ROOT}"
```

Reply:
@AndreyPavlenko  
`-p` flag will do nothing if the directory is existing

Reply:
If the script is running with -e option, it will fail when the directory exists, because the exit code of mkdir will not be 0.

Reply:
@AndreyPavlenko 


> If the script is running with -e option, it will fail when the directory exists, because the exit code of mkdir will not be 0.

I tried on my local and the exit code is zero if `-p` flag is set for mkdir.

Reply:
You are right and it's mentioned in the doc:
```
       -p, --parents
              **no error if existing**, make parent directories as needed, with their file modes unaffected by any -m option.
```